includes/MediaWiki/Request/EditRequest.php

```diff
@@ -2,10 +2,36 @@
 
 namespace MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request;
 
-interface EditRequest {
+use Wikibase\DataModel\Entity\Item;
+use Wikibase\DataModel\Entity\PropertyId;
 
-	public function entity() : ?array;
+/**
+ * @license GPL-2.0-or-later
+ */
+class EditRequest {
 
-	public function reconcile() : ?array;
+	/** @var Item */
+	private $entity;
+
+	/** @var PropertyId */
+	private $reconcilePropertyId;
+
+	public function __construct( Item $entity, PropertyId $reconcilePropertyId ) {
+		$this->entity = $entity;
+		$this->reconcilePropertyId = $reconcilePropertyId;
+	}
+
+	public function entity(): Item {
+		return $this->entity;
+	}
+
+	/**
+	 * The property ID that should be used for reconciliation.
+	 *
+	 * The property is guaranteed to have the datatype "url".
```

Comment:
Maybe this should point to the EditRequestParser as this claim is not guaranteed by this class.

Reply:
Hm, I’m not sure about that, since the `EditRequest` class could also be instantiated somewhere else… how about mentioning it in the constructor phpdoc?

```php
 * @param PropertyId $reconcilePropertyId Must have the datatype "url".
 ```

Reply:
ok :+1:

Reply:
Alright, added as another commit.